docs/contributing.rst

```diff
@@ -46,6 +46,154 @@ a human. All code style enforcement must be automated to avoid bikeshedding
 and losing time.
 
 
+Release Notes
+=======================
+Release notes are the primary product documentation a user will see when updating the library. Therefore, we must take care to ensure the quality of release notes.
+
+A release note entry should be included for every pull request that changes how a user interacts with the library.
+
+Requiring a Release Note
+++++++++++++++++++++++++
+
+A release note is **required** if a PR is user-impacting, or if it meets any of the following conditions:
+
+* `Breaking change to the public API <https://ddtrace.readthedocs.io/en/stable/versioning.html#release-versions>`_
+* New feature
+* Bug fix
+* Deprecations
+* Dependency upgrades
+
+Otherwise, a release note is not required.
+Examples of when a release note is **not required** are:
+
+* CI chores (e.g., upgrade/pinning dependency versions to fix CI)
+* Changes to internal API (Non-public facing, or not-yet released components/features)
+
+Release Note Style Guidelines
++++++++++++++++++++++++++++++
+
+The main goal of a release note is to provide a brief overview of a change.
+If necessary, we can also provide actionable steps to the user.
+
+The release note should clearly communicate what the change is, why the change was made,
+and how a user can migrate their code.
+
+Release notes should:
+
+* Use plain language
+* Avoid overly technical language
+* Be concise
+* Include actionable steps with the necessary code changes
+* Include relevant links (bug issues, upstream issues or release notes, documentation pages)
+* Be full sentences with proper punctuation.
+* Before using specifically acronyms and terminology specific to Datadog, a release note must first introduce them with a definition or explanation.
+
+Release notes should not:
+
+* Be vague. Example: ``fixes an issue in tracing``.
+* Use acronyms liberally. If an acronym must be used, the acronym should be introduced in the first occurrence of the full term in parentheses immediately after the full term.
+* Use dynamic links (``stable/latest/1.x`` URLs). Instead, use static links (specific version, commit hash/tag) whenever possible so that they don't break in the future.
```

Comment:
Is this specific to links to Git repositories?

Reply:
Yes? I think it applies for other cases but we usually only deal with git version-controlled repositories/documentation.